### DISCUSSION TITLE
USRP cannot use external GPSDO as reference clock

### DISCUSSION THREAD

**Discussion by M818M** [srsran/srsRAN_Project#303] [General Help]
Hi,

I am using USRP N210 and LEO Bodnar GPSDO.
As the external clock reference, the GPSDO has successfully found the GPS signal and been locked. ( two LEDs on the front of GPSDO are red and constantly on.)
![image](https://github.com/srsran/srsRAN_Project/assets/149110707/040e9610-b1b5-4bfd-80ed-bc2b4e7c72d3)

I have changed usrp clock source to external.
![image](https://github.com/srsran/srsRAN_Project/assets/149110707/8b14e54c-4f37-4246-aa5d-53fbd7d64828)
and got error
![image](https://github.com/srsran/srsRAN_Project/assets/149110707/8d3d22f2-ce93-4222-ae31-5c998ae15983)
 
When I probe the usrp (using the UHD tool) it tells me that there is no gpsdo module :
![image](https://github.com/srsran/srsRAN_Project/assets/149110707/fc2e18b7-b8a8-4658-843f-b2d57a27aad2)
Is it normal? It seems like the usrp doesn't use the external clock reference. How can I fix it?

**Comment by fllay** [comment]
you can fix by using this one https://www.ettus.com/all-products/gpsdo-mini/ . the one that you use is for external clock reference.

**Reply by pgawlowicz** [reply]
Indeed, "clock_source=gpsdo" is for the internal GPSDO module. With LEO Bodnar GPSDO, you should specify "clock_source=external". However, I do not think that one can use LEO Bodnar GPSDO as a time source, as it does not expose GPS time, nor PPS signal.

**Comment by matildecosta** [comment]
Hi everyone!
When using this setup, the LEO Bodnar Precision GPS Reference output is only the10MHz? Does the board need the PPS also? What to specify in "time_source"?
